pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -484,6 +485,9 @@ func (m *kubeGenericRuntimeManager) readLastStringFromContainerLogs(path string)
 	value := int64(kubecontainer.MaxContainerTerminationMessageLogLines)
 	buf, _ := circbuf.NewBuffer(kubecontainer.MaxContainerTerminationMessageLogLength)
 	if err := m.ReadLogs(context.Background(), path, "", &v1.PodLogOptions{TailLines: &value}, buf, buf); err != nil {
+		if os.IsNotExist(pkgerrs.Cause(err)) {
+			return ""
```

Comment:
did you mean to return `nil`?

Reply:
I don't want to return `nil`. The method `readLastStringFromContainerLogs(path string) string` is returning string, and does not allow return `nil`.

Reply:
Almost everything makes sense to me but the fact that I'm a bit worried we can silently swallow exceptional cases on which there is no log at all - not just errors we rarely hit on races.
Do we have options to avoid this to happen?
Besides that lgtm.

Reply:
Currently, I am thinking this kind of race is allowed and those exceptional cases are not causing other issues. If we know how to signal those rare cases we can have follow-up PR to avoid it to happen.

Reply:
The problem I see is that before this PR we log too much: we report a false positive error when we should not. This is indeed annoying. OTOH now we possibly swallow much rarer but also much more severe error conditions.
However adding state to track this error seems probably too much.